### DISCUSSION TITLE
Build wtih jekyll

### DISCUSSION THREAD

**Discussion by mhaemnn** [alshedivat/al-folio#2928] [Q&A]
I didn't pork, but downloaded it directly and made the theme into a github page. but when I do a push there is always an error:

1. error
Error:  Logging at level: debug Configuration file: /github/workspace/./_config.yml GitHub Pages: github-pages v232 GitHub Pages: jekyll v3.10.0 Theme: jekyll-theme-primer Theme source: /usr/local/bundle/gems/jekyll-theme-primer-0.6.0 Requiring: jekyll-github-metadata Requiring: jekyll-seo-tag Requiring: jekyll-feed Requiring: jekyll-sitemap Requiring: jemoji Requiring: jekyll-coffeescript Requiring: jekyll-commonmark-ghpages Requiring: jekyll-gist Requiring: jekyll-github-metadata Requiring: jekyll-paginate Requiring: jekyll-relative-links Requiring: jekyll-optional-front-matter Requiring: jekyll-readme-index Requiring: jekyll-default-layout Requiring: jekyll-titles-from-headings GitHub Metadata: Initializing... Source: /github/workspace/. Destination: /github/workspace/./_site Incremental build: disabled. Enable with --incremental Generating... EntryFilter: excluded /bin EntryFilter: excluded /Gemfile EntryFilter: excluded /Gemfile.lock Reading: _posts/2015-03-15-formatting-and-links.md Globbed Scope Path: assets/bibliography/2018-12-22-distill.bib Globbed Scope Path: assets/css/main.scss Globbed Scope Path: assets/img/1.png Globbed Scope Path: assets/img/10.jpg Globbed Scope Path: assets/img/11.jpg Globbed Scope Path: assets/img/2.png Globbed Scope Path: assets/img/3.png Globbed Scope Path: assets/img/4.jpg Globbed Scope Path: assets/img/5.jpg Globbed Scope Path: assets/img/6.jpg Globbed Scope Path: assets/img/7.jpg Globbed Scope Path: assets/img/8.jpg Globbed Scope Path: assets/img/9.jpg Globbed Scope Path: assets/img/prof_pic.jpg Globbed Scope Path: assets/img/publication_preview/Simultaneity.gif Globbed Scope Path: assets/img/publication_preview/VW8E.gif Globbed Scope Path: assets/img/publication_preview/brownian-motion.gif Globbed Scope Path: assets/img/publication_preview/expansion_engine.gif Globbed Scope Path: assets/img/publication_preview/wave-mechanics.gif Globbed Scope Path: assets/js/common.js Globbed Scope Path: assets/js/dark_mode.js Globbed Scope Path: assets/js/distillpub/overrides.js Globbed Scope Path: assets/js/distillpub/template.v2.js Globbed Scope Path: assets/js/distillpub/template.v2.js.map Globbed Scope Path: assets/js/distillpub/transforms.v2.js Globbed Scope Path: assets/js/distillpub/transforms.v2.js.map Globbed Scope Path: assets/js/masonry.js Globbed Scope Path: assets/js/theme.js Globbed Scope Path: assets/js/zoom.js Globbed Scope Path: assets/pdf/example_pdf.pdf Globbed Scope Path: assets/plotly/demo.html Globbed Scope Path: assets/bibliography/2018-12-22-distill.bib Globbed Scope Path: assets/css/main.scss Globbed Scope Path: assets/img/1.png Globbed Scope Path: assets/img/10.jpg Globbed Scope Path: assets/img/11.jpg Globbed Scope Path: assets/img/2.png Globbed Scope Path: assets/img/3.png Globbed Scope Path: assets/img/4.jpg Globbed Scope Path: assets/img/5.jpg Globbed Scope Path: assets/img/6.jpg Globbed Scope Path: assets/img/7.jpg Globbed Scope Path: assets/img/8.jpg Globbed Scope Path: assets/img/9.jpg Globbed Scope Path: assets/img/prof_pic.jpg Globbed Scope Path: assets/img/publication_preview/Simultaneity.gif Globbed Scope Path: assets/img/publication_preview/VW8E.gif Globbed Scope Path: assets/img/publication_preview/brownian-motion.gif Globbed Scope Path: assets/img/publication_preview/expansion_engine.gif Globbed Scope Path: assets/img/publication_preview/wave-mechanics.gif Globbed Scope Path: assets/js/common.js Globbed Scope Path: assets/js/dark_mode.js Globbed Scope Path: assets/js/distillpub/overrides.js Globbed Scope Path: assets/js/distillpub/template.v2.js Globbed Scope Path: assets/js/distillpub/template.v2.js.map Globbed Scope Path: assets/js/distillpub/transforms.v2.js Globbed Scope Path: assets/js/distillpub/transforms.v2.js.map Globbed Scope Path: assets/js/masonry.js Globbed Scope Path: assets/js/theme.js Globbed Scope Path: assets/js/zoom.js Globbed Scope Path: assets/pdf/example_pdf.pdf Globbed Scope Path: assets/plotly/demo.html Globbed Scope Path: assets/bibliography/2018-12-22-distill.bib Globbed Scope Path: assets/css/main.scss Globbed Scope Path: assets/img/1.pn

3. warning 
Warning: The github-pages gem can't satisfy your Gemfile's dependencies. If you want to use a different Jekyll version or need additional dependencies, consider building Jekyll site with GitHub Actions: https://jekyllrb.com/docs/continuous-integration/github-actions/
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
  Liquid Exception: Liquid syntax error (/github/workspace/_includes/selected_papers.html line 4): Unknown tag 'bibliography' included in /_layouts/about.html
/usr/local/bundle/gems/liquid-4.0.4/lib/liquid/document.rb:23:in `unknown_tag': Liquid syntax error (/github/workspace/_includes/selected_papers.html line 4): Unknown tag 'bibliography' included  (Liquid::SyntaxError)
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/document.rb:11:in `block in parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:32:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/document.rb:10:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/document.rb:5:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:130:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:114:in `parse'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:13:in `block in parse'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:49:in `measure_time'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:12:in `parse'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/tags/include.rb:174:in `load_cached_partial'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/tags/include.rb:135:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:103:in `render_node_to_output'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:91:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:46:in `block (2 levels) in render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:44:in `each'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:44:in `block in render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/context.rb:123:in `stack'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:43:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:103:in `render_node_to_output'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:82:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:206:in `block in render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:240:in `with_profiling'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:205:in `render'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:218:in `render!'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:30:in `block (2 levels) in render!'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:42:in `measure_bytes'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:29:in `block in render!'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:49:in `measure_time'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:28:in `render!'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:126:in `render_liquid'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:195:in `render_layout'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:158:in `place_in_layouts'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:88:in `render_document'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:62:in `run'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:479:in `render_regenerated'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:472:in `block in render_pages'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:471:in `each'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:471:in `render_pages'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:192:in `render'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:71:in `process'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/command.rb:28:in `process_site'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/commands/build.rb:65:in `build'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/commands/build.rb:36:in `process'
	from /usr/local/bundle/gems/github-pages-232/bin/github-pages:70:in `block (3 levels) in <top (required)>'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
	from /usr/local/bundle/gems/github-pages-232/bin/github-pages:6:in `<top (required)>'
	from /usr/local/bundle/bin/github-pages:25:in `load'
	from /usr/local/bundle/bin/github-pages:25:in `<main>'

**Comment by george-gca** [comment]
You should follow the install instructions as to how setup the site.

**Reply by mhaemnn** [reply]
I've watched the video tutorial, so what if the repository has been created before, and I want to use al-folio v.8?

**Reply by george-gca** [reply]
v.8 is from way behind, so probably a few things won't work out if you are using docker manually. Maybe if you are using the devcontainers or local installation solution it will work fine.
